products/caasp/main.pm

```diff
@@ -68,7 +68,7 @@ sub load_boot_tests {
 # One-click installer - fate#322328
 sub load_inst_tests {
     if (get_var 'AUTOYAST') {
-        loadtest 'autoyast/installation';
+        loadtest 'autoyast/installation' unless check_var('FLAVOR', 'CaaSP-DVD-Incidents');
```

Comment:
It's not very clear here why QAM tests should be different here so without I good comment in the code I would not recommend to do that.

Reply:
this is weird, what about check move to line 70,something like  `if  if (get_var 'AUTOYAST' && ! check_var('FLAVOR', 'CaaSP-DVD-Incidents'))

?

Reply:
@mimi1vx i agree with you the IF is weird, but if that is done, then the else needs to be a `elsif(! getvar 'AUTOYAST')`.

I think it will be easier to read if the order is changed. The code in the ELSE becames the first IF. Example:

`
if (! get_var 'AUTOYAST') {
...
....
} elsif(!check_var('FLAVOR', 'CaaSP-DVD-Incidents')) {
....
}
`
This way we avoid using `&&` in the first condition :-)

Just my 2cents.

Reply:
the testing plan for CaaSP in QAM is quite liquid atm: constantly changing every week, thus I would prefer to not refactor anything until it's gets solid. I will file a poo with refactoring the `main.pm` and writing some documentation for the parallel workers (*what triggers what, and what waits for what to finish*). 

As for this particular code-snippet, my intention is to be less invasive. I am just *disabling* a test for qam: `if` the `autoyast` parameter is present, then `run QA-CSS` tests `unless this is QAM` . The purpose of `main.pm` is just a manuscript for understanding the order of things, not the beauty/structure of the code. Maybe this `if` looks weird, but changing it -- as @dasantiago  pointed out -- it requires to change the original statement into a negative one, which makes things even more weirder.

Reply:
I agree with @okurz here, this issue is not QAM specific.

I see that configuration of QAM cluster is very generous with resources - 4CPU / 8GB RAM for every node - my guess is that you are running out of CPUS and QEMU is not prioritizing the way it should.

I would try to use QA configuration (1CPU/4GB RAM for nodes, 4CPU/8GB RAM for admin), if that does not help we can try limit number of workers per machine. It should help also with other issues I see in QAM group.